src/customer/strategies/googlepay/googlepay-authorizenet-customer-strategy.spec.ts

```diff
@@ -0,0 +1,258 @@
+    let requestSender: RequestSender;
+    let store: CheckoutStore;
+    let strategy: CustomerStrategy;
+    let walletButton: HTMLAnchorElement;
+
+    beforeEach(() => {
+        paymentMethod = getPaymentMethod();
+
+        store = createCheckoutStore({
+            checkout: getCheckoutState(),
+            customer: getCustomerState(),
+            config: getConfigState(),
+            cart: getCartState(),
+            paymentMethods: getPaymentMethodsState(),
+        });
+
+        requestSender = createRequestSender();
+
+        remoteCheckoutActionCreator = new RemoteCheckoutActionCreator(
+            new RemoteCheckoutRequestSender(requestSender)
+        );
+
+        paymentProcessor = createGooglePayPaymentProcessor(
+            store,
+            new GooglePayAuthorizeNetInitializer()
+        );
+
+        formPoster = createFormPoster();
+
+        strategy = new GooglePayCustomerStrategy(
+            store,
+            remoteCheckoutActionCreator,
+            paymentProcessor,
+            formPoster
+        );
+
+        jest.spyOn(store, 'dispatch')
+            .mockReturnValue(Promise.resolve(store.getState()));
+
+        jest.spyOn(paymentProcessor, 'initialize')
+            .mockReturnValue(Promise.resolve());
+
+        jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+            .mockReturnValue(paymentMethod);
+
+        jest.spyOn(formPoster, 'postForm')
+            .mockReturnValue(Promise.resolve());
+
+        container = document.createElement('div');
+        container.setAttribute('id', 'googlePayCheckoutButton');
+        walletButton = document.createElement('a');
+        walletButton.setAttribute('id', 'mockButton');
+
+        jest.spyOn(paymentProcessor, 'createButton')
+            .mockReturnValue(walletButton);
+
+        container.appendChild(walletButton);
+        document.body.appendChild(container);
+    });
+
+    afterEach(() => {
+        document.body.removeChild(container);
+    });
+
+    it('creates an instance of GooglePayCustomerStrategy', () => {
+        expect(strategy).toBeInstanceOf(GooglePayCustomerStrategy);
+    });
+
+    describe('#initialize()', () => {
+
+        describe('Payment method exist', () => {
+
+            it('Creates the button', async () => {
+                customerInitializeOptions = getAuthNetCustomerInitializeOptions();
+
+                await strategy.initialize(customerInitializeOptions);
+
+                expect(paymentProcessor.createButton).toHaveBeenCalled();
+            });
+
+            it('fails to initialize the strategy if no GooglePayCustomerInitializeOptions is provided ', () => {
+                customerInitializeOptions = getAuthNetCustomerInitializeOptions(Mode.Incomplete);
+
+                function initialize() {
+                    strategy.initialize(customerInitializeOptions);
+                }
+
+                expect(initialize).toThrow(InvalidArgumentError);
```

Comment:
you dont need to define a function, you could simply do
```js
expect(() => strategy.initialize(customerInitializeOptions))
  .toThrow(InvalidArgumentError);
```

Reply:
Ok, I just wanted to be more explicit.